components/Step4/Step4.tsx

```diff
@@ -0,0 +1,118 @@
+import React, {useEffect, useState} from 'react';
+import {Pressable, ScrollView, Text, View} from 'react-native';
+import {Input} from '../../components';
+import {selectWizard, updateUserData} from '../../redux/reducers/WizardData';
+import {useDispatch, useSelector} from 'react-redux';
+import {fitnessWorkouts} from '../../utils/fitnessConstants';
+import {FontAwesomeIcon} from '@fortawesome/react-native-fontawesome';
+import {faCheck} from '@fortawesome/free-solid-svg-icons';
+import {Colors} from '../../utils/colors';
+import {wizard} from '../../utils/constants';
+
+import globalStyle from '../../assets/styles/globalStyle';
+import style from './style';
+
+const Step4 = () => {
+  const dispatch = useDispatch();
+  const [customWorkoutValue, setCustomWorkoutValue] = useState('');
+  const wizardData = useSelector(selectWizard);
+  const [fitnessCategoriesWorkout, setFitnessCategoriesWorkout] = useState<
+    Array<{label: string; checked: boolean}>
+  >([...fitnessWorkouts]);
+
+  useEffect(() => {
+    let unselectedItems: Array<{label: string; checked: boolean}>;
+
+    if (wizardData.workouts.length) {
+      unselectedItems = fitnessCategoriesWorkout.filter(
+        category =>
+          wizardData.workouts.findIndex(
+            workout => workout.label === category.label,
+          ) === -1,
+      );
+    } else {
+      unselectedItems = [...fitnessWorkouts];
+    }
+
+    setFitnessCategoriesWorkout([...wizardData.workouts, ...unselectedItems]);
+  }, []);
+
+  const onBlurFitnessInput = (value: string) => {
+    if (
+      value &&
+      !fitnessCategoriesWorkout.filter(
+        workout =>
+          workout.label.toLocaleLowerCase() === value.toLocaleLowerCase(),
+      ).length
+    ) {
+      setFitnessCategoriesWorkout([
+        {label: value, checked: true},
+        ...fitnessCategoriesWorkout,
+      ]);
+
+      dispatch(
+        updateUserData({
+          workouts: [...wizardData.workouts, {label: value, checked: true}],
+        }),
+      );
+    }
+  };
+
+  const onPressWorkoutItem = (workout: {label: string; checked: boolean}) => {
+    const changedCategory = {label: workout.label, checked: !workout.checked};
+    const changedWorkouts = fitnessCategoriesWorkout.map(category =>
+      category.label === workout.label ? changedCategory : category,
+    );
+    setFitnessCategoriesWorkout([...changedWorkouts]);
+
+    dispatch(
+      updateUserData({
+        workouts: changedWorkouts.filter(category => category.checked),
+      }),
+    );
+  };
+
+  return (
+    <View style={[globalStyle.marginTop30, globalStyle.flex]}>
+      <View style={globalStyle.input}>
+        <Input
```

Comment:
To improve the UX, you'd have to register this input and add some sort of validation like `yup` or `zod`